src/server/mod.rs

```diff
@@ -105,13 +112,33 @@ impl<B: AsRef<[u8]> + 'static> Http<B> {
 
         Ok(Server {
             new_service: new_service,
-            core: core,
+            internal: ServerInternal::Core(core),
             listener: listener,
             protocol: self.clone(),
             shutdown_timeout: Duration::new(1, 0),
         })
     }
 
+    /// Bind the provided `addr` and return a server that has
+    /// a shared `Core`.
+    /// 
+    /// This method, when provided a handle to a `Core`, allows
+    /// the ability to bind multiple servers to the same `Core`.
+    pub fn bind_handle<S, Bd>(&self, addr: &SocketAddr, new_service: S, handle: Handle) -> Server<S, Bd>
+        where S: NewService<Request = Request, Response = Response<Bd>, Error = ::Error> + 'static,
+            Bd: Stream<Item=B, Error=::Error>,
+    {
+        let listener = TcpListener::bind(addr, &handle).unwrap();
```

Comment:
This probably means `bind_handle` should return a `Result`, just like `bind` does, so that we don't panic.